### DISCUSSION TITLE
Videos to record

### DISCUSSION THREAD

**Discussion by pokey** [cursorless-dev/cursorless#506] [Ideas]
Starting a discussion to get feedback on videos I'm planning to record.  Please upvote the ideas you're interested in, and use the comments on each item for clarification questions.  Also feel free to propose ideas as new free-standing comments at the bottom of this page

**Comment by pokey** [comment]
More what’s next videos in the style of https://youtu.be/OclW2F8t68U, maybe looking more at projects and milestones

**Comment by pokey** [comment]
Target inference from a user perspective, including the "just" modifier

**Comment by pokey** [comment]
- Install hud and walk through #360

**Comment by pokey** [comment]
- #482

**Comment by pokey** [comment]
- #361

**Comment by pokey** [comment]
- #425

**Comment by pokey** [comment]
Longer form narrated coding video

**Comment by pokey** [comment]
More short, narrated coding sessions

**Comment by pokey** [comment]
More non-narrated coding sessions like https://youtu.be/8dddWzR8AUc

**Reply by ym-han** [reply]
pasting in my ideas from the Slack:

it’d also be nice to see an optimized workflow for things like: 

- navigating and working through a large code base that one may not be familiar with, 
- cross-referencing definitions/declarations on the fly (though then again there may not be very much to optimize here), 
- looking up docs on the fly (this is partly a window management workflow question), 
- working with the debugger and build tools 
- misc dev-op-y or remote server related things (that one usually does while developing software)

**Reply by pokey** [reply]
Given that you've left this one as a comment on "More non-narrated coding sessions", does that mean you'd like these to be non-narrated coding sessions?

**Reply by ym-han** [reply]
> Given that you've left this one as a comment on "More non-narrated coding sessions", does that mean you'd like these to be non-narrated coding sessions?

Ah I think I'd just assumed that non-narrated coding sessions would be the easiest for you to do. But if there's time, it'd probably be more pedagogically effective to add some explanation (either in writing or in video narrated form, whichever is easier). And the best approach, pedagogically speaking, would probably be to have a mix of both short and longer videos, though it's going to depend a lot on factors like 

(i) the nature of the task in question --- window management might be adequately treated in shorter videos, whereas for navigating and working through a large, new codebase, there's a lot of value in seeing how an expert would approach it in a more unscripted way
(ii) whether there are sequences that would make sense to extract and make a shorter video out of --- e.g. if the sequence is complicated, or if it seems like the kind of thing someone would want to use pretty often but wouldn't stumble upon independently

**Comment by pokey** [comment]
Clone and modify in one step (eg `"take air clone line word hello"`)

**Comment by pokey** [comment]
Cursorless test case recorder, including video of running tests, recording a test, and showing recorded test yaml

**Comment by pokey** [comment]
easy-level leetcode problems (thanks for the idea @mzizzi!)

**Comment by pokey** [comment]
Parrot training and setup

**Reply by pokey** [reply]
Done! ✅ https://www.youtube.com/watch?v=2j90vhzRoT8&list=PLkafpFOBVedTV8E7Lfw4hSH2oAtPyjjl1

**Comment by pokey** [comment]
How to add a new task with associated voice command for common shell commands

**Comment by pokey** [comment]
Adding sleeps in your voice commands to increase chaining ability

**Comment by pokey** [comment]
Customization (basically walk through of [customization.md](https://github.com/pokey/cursorless-talon/blob/main/docs/customization.md))

**Comment by pokey** [comment]
HTML support

**Comment by pokey** [comment]
Feature highlight videos for specific actions

**Comment by pokey** [comment]
How cursorless continuous delivery and monorepo works

Note to self: use https://man.archlinux.org/man/git-subtree.1 as reference for git subtree

**Comment by pokey** [comment]
The new `"paint"` scope type and `"shuffle"` action

**Comment by pokey** [comment]
"bring" / "move"

**Comment by pokey** [comment]
Multiple cursors

**Comment by pokey** [comment]
CSV editing workflow, including multiple cursors

- Include “move line before air” when cursors in multiple lines to move them all next to each other

**Comment by pokey** [comment]
Curating a [zettelkasten](https://g.co/kgs/Lx7T9S) using [foam](https://foambubble.github.io/foam/) and Cursorless

**Reply by ym-han** [reply]
just to add: i'd also be interested in figuring out a workflow for incorporating spaced repetition software like anki as well (see, e.g., https://emacsconf.org/2021/talks/cs/ for one such workflow with emacs)

**Comment by pokey** [comment]
Simple talon scripting tutorials, eg creating a list, a capture, etc

**Reply by pokey** [reply]
One version of this video: walk through the process of adding a single feature that requires me to use capture list, etc

**Reply by pokey** [reply]
another version of this video (also useful): small, eg 2 min videos, each covering a single Talon concept, eg capture, list, etc

**Reply by pokey** [reply]
(see also https://github.com/cursorless-dev/cursorless/discussions/506#discussioncomment-4896501)

**Comment by soheeyang** [comment]
It would be great if you could upload the raw json file as well as the formatted HTML page. (I actually forgot what I suggested earlier, but I guess it was this one 😅)

**Reply by pokey** [reply]
Fwiw they currently live here https://github.com/pokey/pokey.github.io/tree/main/src/data/videos

**Comment by soheeyang** [comment]
Ohh, it was actually this one: please upload some videos of you coding naturally, switching back and forth between the browser, terminal, and code editor, in order to show people how you use Talon (I learned a lot from how you generally use Talon). There can be some gaps in between where you say nothing, but I think that's just fine.

**Reply by pokey** [reply]
Here you go!
https://youtu.be/Qg11LG0738c

**Reply by soheeyang** [reply]
Thank you so much! 🎉

**Comment by pokey** [comment]
Video where I do the same thing by keyboard and by voice

**Comment by pokey** [comment]
Super quick six second videos of quick actions

**Comment by pokey** [comment]
"take token" ie ability to select adjacent token and how it decides which to select

**Comment by pokey** [comment]
Sub-token scope types, eg "chuck last char air"

**Comment by pokey** [comment]
How to run cursorless extension in debug mode

**Comment by pokey** [comment]
Merging your talon files with knausj using [imerge](https://github.com/mhagger/git-imerge) in VSCode

**Comment by HermanPerkins** [comment]
A tutorial video for noncoders that uses plain prose and doesn't use any sort of language that might be too confusing! Maybe focusing just on the colors/commands

**Comment by pokey** [comment]
Tiktok-style video where I start with the following:

```
this is some other text
whatever this is some text
whatever this is some other text
whatever this is some text
this is some text
whatever this is some text
this is some text
whatever this is some text
whatever this is some text
this is some text
whatever this is some text


This is where I want to move it
```

And then say `"take whale breed all move line after trap"`

**Comment by pokey** [comment]
Single commands

**Reply by pokey** [reply]
Did a nice chain using new composable modifiers.  I wanted to sort the languages on the RHS of this assignment in `Make`:

```make
LANGUAGES = agda c clojure cpp css c-sharp bash go haskell html java javascript json latex markdown php python ruby rust scala scss sparql tsx typescript yaml
```

The command was `“sort every paint tail gust”` (where `“gust”` referred to `agda`)

_Edit_: Done! https://youtu.be/EwdpPN-21-g

**Comment by pokey** [comment]
Tricks to break up long commands.  Eg

- "take air and bat" followed by "take this and cap and drum"
- "copy air and bat" followed by "paste to cap and drum"
- etc
- Speech timeout

**Comment by pokey** [comment]
How to tweak your hats for better readability

**Comment by AndreasArvidsson** [comment]
Videos on Talon concepts/code

Short videos explaining each concept with code example
* What is an action
* What is a module
* What is a context
* What is a capture
* What is a list

A longer video showing from beginning to end how you add new commands with different context captures and so on. Maybe add support for a new application?

**Reply by pokey** [reply]
duplicate of https://github.com/cursorless-dev/cursorless/discussions/506#discussioncomment-2211226

**Comment by pokey** [comment]
An opinionated video on how to use dictation mode

**Comment by pokey** [comment]
VSCode basics
- "pop sesh"
- "file hunt <phrase>"
- "bar switch"
- "scout"
- "scout all"
